include/geode/geometry/aabb.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2019 - 2020 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+/*
+ * Modified from RINGMesh https://github.com/ringmesh/RINGMesh
+ * Copyright (c) 2012-2018, Association Scientifique pour la Geologie et ses
+ * Applications (ASGA)
+ */
+
+#pragma once
+
+#include <absl/types/span.h>
+
+#include <geode/basic/pimpl.h>
+
+#include <geode/geometry/bounding_box.h>
+#include <geode/geometry/common.h>
+
+namespace geode
+{
+    template < index_t dimension >
+    class AABBTree
+    {
+        OPENGEODE_DISABLE_COPY_AND_MOVE( AABBTree );
+        OPENGEODE_TEMPLATE_ASSERT_2D_OR_3D( dimension );
+
+    public:
+        /*!
+         * AABB tree structure implementation
+         * The tree is store in s single vector following this example:
```

Comment:
I would remove this comment since it is now an implementation detail. What do you think ?

Reply:
I think that maybe this comment is not well suited here however it helps to understand what the aabb is building and how to use it. Maybe we should change it a bit but i think it worth to keep something.

Reply:
You are not suppose to care how it is built! Not in the interface. The same comment is in the aabb_impl.h but in aabb.h, I would remove it. @panquez what do you think?

Reply:
I agree with Arnaud that the comment should be tell how it works but what is does. For several reason: generally you are more interested on what this the contract of a class than the way it reached this contract, if the implementation changes without changing the contract (for performance or something else), it should be transparent for the user. Moreover, if implementation changes in implementation files, this documentation in interface file will diverge surely at some point.

Reply:
So I agree with François that a comment should remains but not this one ;)

Reply:
I changed the documentation.